radicle-node/src/wire/protocol.rs

```diff
@@ -49,12 +47,99 @@ impl<G: Signer + EcSign> fmt::Debug for Control<G> {
     }
 }
 
+#[derive(Debug)]
+pub struct WireRuntime {
+    node: NodeId,
+    state: WireState,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+pub enum WireState {
+    Outbound,
+    Inbound { id: Option<NodeId> },
+}
+
+impl WireRuntime {
+    fn outbound(node: NodeId) -> Self {
+        Self {
+            node,
+            state: WireState::Outbound,
+        }
+    }
+
+    fn inbound(node: NodeId) -> Self {
+        Self {
+            node,
+            state: WireState::Inbound { id: None },
+        }
+    }
+}
+
+impl NetStateMachine for WireRuntime {
+    const NAME: &'static str = "wire";
+
+    type Init = Vec<u8>;
+    type Artifact = NodeId;
+    type Error = io::Error;
+
+    fn init(&mut self, init: Vec<u8>) {
+        println!(">>>>>>>>>> {} INIT", self.node);
+        //
+    }
+
+    fn next_read_len(&self) -> usize {
+        println!(">>>>>>>>>> {} NEXT READ LEN", self.node);
+        match self.state {
+            WireState::Outbound => 0,
+            WireState::Inbound { .. } => 34,
+        }
+    }
+
+    fn advance(&mut self, input: &[u8]) -> Result<Vec<u8>, Self::Error> {
+        panic!();
+        println!(">>>>>>>>>> {} ADVANCE", self.node);
+        match self.state {
+            WireState::Outbound => {
+                let mut buf = Vec::new();
+                let msg = Message::init(self.node);
+                let _ = msg.encode(&mut buf);
+
+                return Ok(buf);
+            }
+            WireState::Inbound { id } if !input.is_empty() => {
+                let msg = Message::decode(&mut io::Cursor::new(input)).unwrap();
+                if let Message::Initialize { node_id } = msg {
+                    println!(">>>>>> GOT INIT MESSAGE {node_id}");
+                }
+                return Ok(vec![]);
+            }
+            _ => {}
+        }
+
+        Ok(vec![])
+    }
+
+    fn artifact(&self) -> Option<Self::Artifact> {
+        println!(">>>>>>>>>> {} ARTIFACT", self.node);
+        None
+        // self.state.remote_cert().cloned()
+    }
+
+    fn is_init(&self) -> bool {
+        println!(">>>>>>>>>> {} IS INIT", self.node);
+        true
```

Comment:
I do not remember what this method should do, but it looks like this `true` is the source why the handshake doesn't progress